x/btcstkconsumer/README.md

```diff
@@ -0,0 +1,341 @@
+Babylon's Bitcoin-staked security to enhance their own economic guarantees.
+
+This module acts as a central registry, maintaining a canonical list of all BSNs, their metadata, and the
+finality providers associated with them. It provides the necessary logic for registering new BSNs,
+querying their information, and enforcing integration policies defined by the Babylon network's governance.
+
+## Table of Contents
+
+1.  [Concepts](#1-concepts)
+*   [1.1. Consumer/BSN](#11-consumerbsn)
+*   [1.2. Consumer Registration](#12-consumerbsn-registration)
+*   [1.3. Consumer Types](#13-consumerbsn-types)
+*   [1.4. Finality Provider Registry](#14-finality-provider-registry)
+*   [1.5. Permissioned vs. Permissionless Integration](#15-permissioned-vs-permissionless-integration)
+2.  [State](#2-state)
+*   [2.1. Parameters](#21-parameters)
+*   [2.2. Consumer Registry](#22-consumer-registry)
+*   [2.3. Finality Provider Mappings](#23-finality-provider-mappings)
+3.  [Messages](#3-messages)
+*   [3.1. `MsgRegisterConsumer`](#31-msgregisterconsumer)
+*   [3.2. `MsgUpdateParams`](#32-msgupdateparams)
+4.  [BeginBlocker / EndBlocker](#4-beginblocker--endblocker)
+5.  [Events](#5-events)
+*   [5.1. `EventConsumerRegistered`](#51-eventconsumerregistered)
+6.  [Queries](#6-queries)
+*   [6.1. Parameters Queries](#61-parameters-queries)
+*   [6.2. Consumer Queries](#62-consumer-queries)
+*   [6.3. Finality Provider Queries](#63-finality-provider-queries)
+
+## 1. Concepts
+
+### 1.1. Consumer/BSN
+
+<!--TO DO: change all instances of consumer to bsn when code changed -->
+
+A **Consumer/BSN** is any external blockchain or rollup that registers with Babylon to utilise its security services. The
+`btcstkconsumer` module maintains a unique record for each BSN.
+
+### 1.2. Consumer/BSN Registration
+
+This is the process by which a new BSN is officially recognised by Babylon. The registration process
+captures essential metadata about the BSN, which varies depending on its integration type. This process is initiated
+via a `MsgRegisterConsumer` transaction.
+
+### 1.3. Consumer/BSN Types
+
+The module supports two primary types of BSN integrations:
+
+#### 1.3.1. Cosmos Consumer/BSN
+
+*   **Description**: A blockchain built with the Cosmos SDK that integrates with Babylon via an IBC channel.
+*   **Identifier (`consumer_id`)**: The IBC Client ID (`e.g., 07-cometbft-0`) that Babylon uses to track the state of
+    the BSN chain.
+*   **Integration Logic**: The module verifies the existence of the specified IBC client on Babylon before completing
+    the registration.
+
+#### 1.3.2. Rollup Consumer/BSN
+
+*   **Description**: A rollup (e.g., an optimistic or ZK-rollup) that uses a smart contract on Babylon for its data
+    availability or finality needs.
+*   **Identifier (`consumer_id`)**: A unique, human-readable string that identifies the rollup (e.g., `my-rollup-chain-1`).
```

Comment:
We can mention that consumer ID for rollup BSNs can be any unique string chosen arbitrarily